csharp/src/Ice/OpaqueEndpointI.cs

```diff
@@ -11,15 +13,20 @@ namespace IceInternal
 {
     internal sealed class OpaqueEndpointI : Endpoint
     {
+        private EndpointType _type;
+        private Ice.Encoding _rawEncoding;
+        private byte[] _rawBytes;
+        private int _hashCode = 0; // 0 is a special value that means not initialized.
+
         public OpaqueEndpointI(List<string> args)
         {
-            _type = -1;
+            _type = (EndpointType)(short)-1;
```

Comment:
Why not add Opaque -1 to the enum?

Reply:
nvm this -1 is to check if the type was initialized

Reply:
We may be able to use a temporary nullable EndpointType in a future refactoring. The current logic does not allow it, since _type is initialized in a round-about manner.